mipengine/controller/controller.py

```diff
@@ -0,0 +1,192 @@
+                return result
+
+        try:
+            algorithm_execution_dto = AlgorithmExecutionDTO(
+                context_id=context_id,
+                algorithm_name=algorithm_name,
+                algorithm_request_dto=algorithm_request_dto,
+            )
+
+            loop = asyncio.get_running_loop()
+
+            # DEBUG(future logging..)
+            print(
+                f"\n(controller.py::exec_algorithm) starts executing-> "
+                f"{algorithm_name=} with {context_id=}\n"
+            )
+            # DEBUG end
+
+            algorithm_result = await loop.run_in_executor(
+                None,
+                run_algorithm_executor_in_threadpool,
+                algorithm_execution_dto,
+                all_nodes_tasks_handlers,
+            )
+
+            # DEBUG(future logging..)
+            print(
+                f"\n(controller.py::exec_algorithm) FINISHED->  {algorithm_name=} "
+                f"with {context_id=}"
+            )
+            print(f"{algorithm_result.json()=}\n")
+            # DEBUG end
+
+            return algorithm_result.json()
+
+        except:
+            logging.error(
+                f"Algorithm execution failed. Unexpected exception: \n "
+                f"{traceback.format_exc()}"
+            )
+
+    def validate_algorithm_execution_request(
+        self, algorithm_name: str, algorithm_request_dto: AlgorithmRequestDTO
+    ):
+        available_datasets_per_schema = self.get_all_available_datasets_per_schema()
+        validate_algorithm_request(
+            algorithm_name=algorithm_name,
+            algorithm_request_dto=algorithm_request_dto,
+            available_datasets_per_schema=available_datasets_per_schema,
+        )
+
+
+    async def start_node_registry(self):
+        asyncio.create_task(node_registry.update())
+
+    async def stop_node_registry(self):
+        node_registry.keep_updating = False
+
+    def get_all_datasets_per_node(self):
+        datasets = {}
+        for node in node_registry.get_all_local_nodes():
+            datasets[node.id] = node.datasets_per_schema
+        return datasets
+
+    def get_all_available_schemas(self):
+        return node_registry.get_all_available_schemas()
+
+    def get_all_available_datasets_per_schema(self):
+        return node_registry.get_all_available_datasets_per_schema()
+
+    def get_all_local_nodes(self):
+        return node_registry.get_all_local_nodes()
+
+    def _create_nodes_tasks_handlers(
+        self, context_id: str, pathology: str, datasets: List[str]
+    ) -> NodesTasksHandlersDTO:
+
+        # Get only teh relevant nodes from the node registry
+        global_node = node_registry.get_all_global_nodes()[0]
+        local_nodes = node_registry.get_nodes_with_any_of_datasets(
+            schema=pathology,
+            datasets=datasets,
+        )
+
+        # Global Node, gather the Celery Parameters
+        global_node_celery_params_dto = CeleryParamsDTO(
```

Comment:
If we use the common "celery_app" module methods for celery instantiation we won't need to read all the parameters from the config each time. All the "fixed" celery configs will be hidden in that method.

Reply:
[MIP-253](https://team-1617704806227.atlassian.net/browse/MIP-253)